src/helpers/base64.js

```diff
@@ -1,84 +1,83 @@
-var base64 = require('base64-js');
+import base64 from 'base64-js';
 
-function padding(str) {
-  var mod = (str.length % 4);
+export function padding(str) {
+  var mod = str.length % 4;
   var pad = 4 - mod;
 
   if (mod === 0) {
     return str;
   }
 
-  return str + (new Array(1 + pad)).join('=');
+  return str + new Array(1 + pad).join('=');
 }
 
-function byteArrayToString(array) {
-  var result = "";
+export function byteArrayToString(array) {
+  var result = '';
   for (var i = 0; i < array.length; i++) {
     result += String.fromCharCode(array[i]);
   }
   return result;
 }
 
-function stringToByteArray(str) {
+export function stringToByteArray(str) {
   var arr = new Array(str.length);
   for (var a = 0; a < str.length; a++) {
     arr[a] = str.charCodeAt(a);
   }
   return arr;
 }
 
-function byteArrayToHex(raw) {
+export function byteArrayToHex(raw) {
   var HEX = '';
 
   for (var i = 0; i < raw.length; i++) {
     var _hex = raw[i].toString(16);
-    HEX += (_hex.length === 2 ? _hex : '0' + _hex);
+    HEX += _hex.length === 2 ? _hex : '0' + _hex;
   }
 
   return HEX;
 }
 
-function encodeString(str) {
-  return base64.fromByteArray(stringToByteArray(encodeURIComponent(str).replace(/%([0-9A-F]{2})/g, function (match, p1) {
-    return String.fromCharCode('0x' + p1);
-  })))
-  .replace(/\+/g, '-') // Convert '+' to '-'
-  .replace(/\//g, '_'); // Convert '/' to '_';
+export function encodeString(str) {
+  return base64
+    .fromByteArray(
+      stringToByteArray(
+        encodeURIComponent(str).replace(/%([0-9A-F]{2})/g, function(match, p1) {
+          return String.fromCharCode('0x' + p1);
+        })
+      )
+    )
+    .replace(/\+/g, '-') // Convert '+' to '-'
+    .replace(/\//g, '_'); // Convert '/' to '_';
 }
 
-function decodeToString(str) {
+export function decodeToString(str) {
   str = padding(str)
     .replace(/\-/g, '+') // Convert '-' to '+'
     .replace(/_/g, '/'); // Convert '_' to '/'
 
-  return decodeURIComponent(byteArrayToString(base64.toByteArray(str)).split('').map(function (c) {
-    return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
-  }).join(''));
+  return decodeURIComponent(
+    byteArrayToString(base64.toByteArray(str))
+      .split('')
+      .map(function(c) {
+        return '%' + ('00' + c.charCodeAt(0).toString(16)).slice(-2);
+      })
+      .join('')
+  );
 }
 
-function decodeToHEX(str) {
+export function decodeToHEX(str) {
   return byteArrayToHex(base64.toByteArray(padding(str)));
 }
 
-function base64ToBase64Url(base64String) {
+export function base64ToBase64Url(base64String) {
   var SAFE_URL_ENCODING_MAPPING = {
-    "+": "-",
-    "/": "_",
-    "=": ""
+    '+': '-',
+    '/': '_',
+    '=': ''
   };
 
   return base64String.replace(/[+/=]/g, function(m) {
     return SAFE_URL_ENCODING_MAPPING[m];
   });
 }
-
-module.exports = {
```

Comment:
IMO this pattern is easier to read because you have in a single place all the functions that this file exports. Why have you changed it?

Reply:
we had to convert from CJS (`module.exports`) to ESM (`export const` or `export default`).